java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java

```diff
@@ -2318,6 +2205,37 @@ public float readFloat() {
     }
   }
 
+  // Reduce method body for better inline in the caller.
+  @CodegenInvoke
+  public float readFloatOnLE() {
+    int readerIdx = readerIndex;
+    // use subtract to avoid overflow
+    int remaining = size - readerIdx;
+    if (remaining < 4) {
+      throw new IndexOutOfBoundsException(
```

Comment:
Nit: we can reuse [`throwIndexOutOfBoundsException`](https://github.com/apache/incubator-fury/blob/main/java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java#L1853-L1857). `throwIndexOutOfBoundsException` is not executed frequently and should have no impact on inlining.

Reply:
This is temprary , it will be replaced by `streamReader.fillBuffer` in #1451. The exception will be trown in `DoundChecker`